hoomd/md/pytest/test_angle.py

```diff
@@ -0,0 +1,185 @@
+    ]
+    cosinesq_arg_list = [
+        dict(zip(_cosinesq_args, val)) for val in zip(*_cosinesq_args.values())
+    ]
+    angle_and_args = []
+    for args in harmonic_arg_list:
+        angle_and_args.append((hoomd.md.angle.Harmonic, args))
+    for args in cosinesq_arg_list:
+        angle_and_args.append((hoomd.md.angle.Cosinesq, args))
+    return angle_and_args
+
+
+def get_angle_args_forces_and_energies():
+    harmonic_arg_list = [
+        dict(zip(_harmonic_args, val)) for val in zip(*_harmonic_args.values())
+    ]
+    cosinesq_arg_list = [
+        dict(zip(_cosinesq_args, val)) for val in zip(*_cosinesq_args.values())
+    ]
+    harmonic_forces = [-1.5708, 2.6180, 2.6180]
+    harmonic_energies = [0.4112, 0.3427, 0.6854]
+    cosinesq_forces = [-1.29904, 1.7936, 1.58494]
+    cosinesq_energies = [0.375, 0.214466, 0.334936]
+
+    angle_args_forces_and_energies = []
+    for i in range(3):
+        angle_args_forces_and_energies.append(
+            (hoomd.md.angle.Harmonic, harmonic_arg_list[i], harmonic_forces[i],
+             harmonic_energies[i]))
+    for i in range(3):
+        angle_args_forces_and_energies.append(
+            (hoomd.md.angle.Cosinesq, cosinesq_arg_list[i], cosinesq_forces[i],
+             cosinesq_energies[i]))
+    return angle_args_forces_and_energies
+
+
+@pytest.fixture(scope='session')
+def triplet_snapshot_factory(device):
+
+    def make_snapshot(d=1.0,
+                      theta_deg=60,
+                      particle_types=['A'],
+                      dimensions=3,
+                      L=20):
+        theta_rad = theta_deg * (np.pi / 180)
+        s = hoomd.Snapshot(device.communicator)
+        N = 3
+        if s.exists:
+            box = [L, L, L, 0, 0, 0]
+            if dimensions == 2:
+                box[2] = 0
+            s.configuration.box = box
+            s.particles.N = N
+            # shift particle positions slightly in z so MPI tests pass
+            s.particles.position[:] = [
+                [-d * np.sin(theta_rad / 2), d * np.cos(theta_rad / 2), 0.1],
+                [0.0, 0.0, 0.1],
+                [d * np.sin(theta_rad / 2), d * np.cos(theta_rad / 2), 0.1]
+            ]
+            s.particles.types = particle_types
+            if dimensions == 2:
+                box[2] = 0
+                s.particles.position[:] = [[
+                    -d * np.sin(theta_rad / 2), d * np.cos(theta_rad / 2) + 0.1,
+                    0.0
+                ], [0.0, 0.1, 0.0],
+                                           [
+                                               d * np.sin(theta_rad / 2),
+                                               d * np.cos(theta_rad / 2) + 0.1,
+                                               0.0
+                                           ]]
+        return s
+
+    return make_snapshot
+
+
+@pytest.mark.parametrize("angle_and_args", get_angle_and_args())
+def test_before_attaching(angle_and_args):
```

Comment:
```suggestion
@pytest.mark.parametrize("angle_cls, potential_kwargs", get_angle_and_args())
def test_before_attaching(angle_cls, potential_kwargs):
```
Make this change in the function and all other functions that use this and similar fixtures in the tests.